runtime/src/main/scala/org/scalaexercises/exercises/Evaluator.scala

```diff
@@ -97,4 +99,15 @@ class Evaluator(timeout: Duration = 20.seconds) {
       }
     }
   }
+
+  def apply[T](pre: String, code: String): EvalResult[T] = {
+    val allCode = s"""
+                     |$pre
+                     |$code
+                   """.stripMargin
+    val fut = Future({ eval(allCode) })
+    Try(
+      Await.result(fut, timeout)
+    ).getOrElse(EvalResult.Timeout())
```

Comment:
Good point! The `EvalResult` returned by the `Future` already carries an exception if it occurs during the execution of the code, the `Await.result` will throw an exception only for the timeouts.